src/vm-memory/src/mmap.rs

```diff
@@ -74,31 +105,32 @@ impl Bytes<MemoryRegionAddress> for GuestRegionMmap {
     type E = guest_memory::Error;
 
     fn write(&self, buf: &[u8], addr: MemoryRegionAddress) -> guest_memory::Result<usize> {
-        // TODO: here be write!
         let maddr = addr.raw_value() as usize;
-        self.local_volatile_slice()
+        let bytes = self
+            .local_volatile_slice()
             .write(buf, maddr)
-            .map_err(Into::into)
+            .map_err(Into::<guest_memory::Error>::into)?;
```

Comment:
Just mentioning this here. There are cases where we get an error, but actually we managed to do a partial write. If this happens I'm not sure whether we should consider the page dirtied or not.